Content.Shared/Interfaces/GameObjects/Components/Interaction/AttackEvent.cs

```diff
@@ -51,15 +32,46 @@ public class AttackEvent : EntityEventArgs
         /// </summary>
         public IEntity? TargetEntity { get; }
 
-        public AttackEvent(IEntity user, EntityCoordinates clickLocation, bool wideAttack, EntityUid target = default)
+        /// <summary>
+        ///     Modified by the handler to indicate whether the attack succeeded.
+        /// </summary>
+        public bool Succeeded { get; set;  }
+
+        public ClickAttackEvent(IEntity user, EntityCoordinates clickLocation, EntityUid target = default)
         {
             User = user;
             ClickLocation = clickLocation;
-            WideAttack = wideAttack;
             Target = target;
 
             IoCManager.Resolve<IEntityManager>().TryGetEntity(Target, out var targetEntity);
             TargetEntity = targetEntity;
         }
     }
+
+    /// <summary>
+    ///     Raised when a target entity is wide attacked by a user.
+    /// </summary>
+    public class WideAttackEvent : EntityEventArgs
+    {
+        /// <summary>
+        ///     Entity that triggered the attack.
+        /// </summary>
+        public IEntity User { get; }
+
+        /// <summary>
+        ///     The original location that was clicked by the user.
+        /// </summary>
+        public EntityCoordinates ClickLocation { get; }
+
+        /// <summary>
+        ///     Modified by the handler to indicate whether the attack succeeded.
+        /// </summary>
+        public bool Succeeded { get; set;  }
```

Comment:
not sure if i'll keep this since its not used by anything and I can't think of a use case for it

Reply:
Presumably it'd be used to prevent other components (after the current component) from consuming the event, which is a common pattern but I'm not 100% sure how much sense it does/doesn't make for attacking/wide attacking.

Reply:
The problem with this is that event handlers don't have a specified order (and they cannot due to implementation details, either) so some handlers might get this event *before* `Succeeded` has been set by the correct handler.
Something like Remie says, a `Handled` property would work much better, I feel. We even have a class you can inherit for that: `HandledEntityEventArgs`

Reply:
decided to just remove it, it can be reimplemented easily if that behavior is needed